Failed webpack build on Yarn v2 with 'Error: Can't resolve 'uuid/v4' in a CRA project on 6.0.0-beta.1

**Describe the bug**

The following error occurs when attempting to build the storybook with `yarn storybook`

```
ERROR in ./.yarn/$$virtual/@storybook-addon-actions-virtual-6af69bc721/0/cache/@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip/node_modules/@storybook/addon-actions/dist/preview/action.js
Module not found: Error: Can't resolve 'uuid/v4' in 'C:\Users\chyyran\coding\hydrogen-paper\theme\.yarn\$$virtual\@storybook-addon-actions-virtual-6af69bc721\0\cache\@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip\node_modules\@storybook\addon-actions\dist\preview'
 @ ./.yarn/$$virtual/@storybook-addon-actions-virtual-6af69bc721/0/cache/@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip/node_modules/@storybook/addon-actions/dist/preview/action.js 10:32-50
 @ ./.yarn/$$virtual/@storybook-addon-actions-virtual-6af69bc721/0/cache/@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip/node_modules/@storybook/addon-actions/dist/preview/index.js
 @ ./.yarn/$$virtual/@storybook-addon-actions-virtual-6af69bc721/0/cache/@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip/node_modules/@storybook/addon-actions/dist/index.js
 @ ./.yarn/$$virtual/@storybook-addon-actions-virtual-6af69bc721/0/cache/@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip/node_modules/@storybook/addon-actions/dist/containers/ActionLogger/index.js
 @ ./.yarn/$$virtual/@storybook-addon-actions-virtual-6af69bc721/0/cache/@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip/node_modules/@storybook/addon-actions/dist/register.js
 @ multi ./.yarn/$$virtual/@storybook-core-virtual-7295b82335/0/cache/@storybook-core-npm-6.0.0-beta.1-859370e4c8-2.zip/node_modules/@storybook/core/dist/server/common/polyfills.js ./.yarn/$$virtual/@storybook-core-virtual-7295b82335/0/cache/@storybook-core-npm-6.0.0-beta.1-859370e4c8-2.zip/node_modules/@storybook/core/dist/client/manager/index.js ./.yarn/$$virtual/@storybook-addon-actions-virtual-6af69bc721/0/cache/@storybook-addon-actions-npm-6.0.0-beta.1-b4989508c0-2.zip/node_modules/@storybook/addon-actions/dist/register.js ./.yarn/$$virtual/@storybook-addon-links-virtual-aaefc1c417/0/cache/@storybook-addon-links-npm-6.0.0-beta.1-56afa7d221-2.zip/node_modules/@storybook/addon-links/dist/register.js
Child HtmlWebpackCompiler:
                          Asset      Size               Chunks  Chunk Names
    __child-HtmlWebpackPlugin_0  7.46 KiB  HtmlWebpackPlugin_0  HtmlWebpackPlugin_0
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [./.yarn/$$virtual/html-webpack-plugin-virtual-1bc9edfd62/0/cache/html-webpack-plugin-npm-4.3.0-a61ded3510-2.zip/node_modules/html-webpack-plugin/lib/loader.js!./.yarn/$$virtual/@storybook-core-virtual-7295b82335/0/cache/@storybook-core-npm-6.0.0-beta.1-859370e4c8-2.zip/node_modules/@storybook/core/dist/server/templates/index.ejs] 2.19 KiB {HtmlWebpackPlugin_0} [built]
```

**To Reproduce**
Steps to reproduce the behavior:
1. `yarn dlx -p @storybook/cli@next sb init` on a CRA project with Yarn v2 with PNP enabled.
2. Ensure `@storybook/addon-actions` is installed and being used.
3. `yarn storybook`

**Expected behavior**
It works and builds

**Code snippets**
My `.yarnrc.yml` for reference
```
packageExtensions:
  "@apollographql/apollo-tools@*":
    dependencies:
      graphql: "*"
  jss-plugin-rule-value-function@*:
    dependencies:
      tiny-warning: "*"
plugins:
  - path: .yarn/plugins/@yarnpkg/plugin-interactive-tools.js
    spec: "@yarnpkg/plugin-interactive-tools"
  - path: .yarn/plugins/@yarnpkg/plugin-typescript.js
    spec: "@yarnpkg/plugin-typescript"

yarnPath: .\.yarn\releases\yarn-berry.js
```

I tried the following addition to `packageExtensions`, to no effect
```
packageExtensions:
  "@storybook/addon-actions@*":
    dependencies:
      uuid: "*"  
```
**System:**
```
Environment Info:

  System:
    OS: Windows 10 10.0.19613
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  Binaries:
    Node: 12.10.0 - ~\AppData\Local\Temp\tmp-213483OYD96zkEdlT\node.CMD
    Yarn: 2.0.0-rc.29 - ~\AppData\Local\Temp\tmp-213483OYD96zkEdlT\yarn.CMD
    npm: 6.10.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.19613.1000.0
  npmGlobalPackages:
    @storybook/cli: 5.3.13
```

--- a/addons/actions/src/preview/action.ts
+++ b/addons/actions/src/preview/action.ts
@@ -1,4 +1,4 @@
-import uuid from 'uuid/v4';
+import { v4 as uuidv4 } from 'uuid';
 import { addons } from '@storybook/addons';
 import { EVENT_ID } from '../constants';
 import { ActionDisplay, ActionOptions, HandlerFunction } from '../models';
@@ -12,7 +12,7 @@ export function action(name: string, options: ActionOptions = {}): HandlerFuncti
 
   const handler = function actionHandler(...args: any[]) {
     const channel = addons.getChannel();
-    const id = uuid();
+    const id = uuidv4();
     const minDepth = 5; // anything less is really just storybook internals
 
     const actionDisplayToEmit: ActionDisplay = {

